src/System.Linq.Parallel/src/System/Linq/Parallel/Enumerables/EmptyEnumerable.cs

```diff
@@ -58,14 +59,14 @@ public override IEnumerator<T> GetEnumerator()
 
     internal class EmptyEnumerator<T> : QueryOperatorEnumerator<T, int>, IEnumerator<T>
     {
-        internal override bool MoveNext(ref T currentElement, ref int currentKey)
+        internal override bool MoveNext([MaybeNull, AllowNull] ref T currentElement, ref int currentKey)
         {
             return false;
         }
 
         // IEnumerator<T> methods.
-        public T Current { get { return default(T); } }
-        object IEnumerator.Current { get { return null; } }
+        public T Current { get { return default!; } }
```

Comment:
It doesn't matter mutch as it  is internal, but for correctness, this should probably have a `[return: MaybeNull]`.

Reply:
i had that at first, but @sharwell commented `It is not valid to widen the return type by adding [MaybeNull], which implemented Enumerator<T>.Current.` https://github.com/dotnet/corefx/pull/41533#discussion_r332178921.

Reply:
I don't understand the feedback, @sharwell.  Either way, this can return null.  With `!`, it breaks the contract but is hidden.  With `[MaybeNull]`, it breaks the contract but at least is visible to anyone consuming this in a strongly-typed manner.